INTEGRATION.md

```diff
@@ -0,0 +1,152 @@
+
+### Gateway from IPFS_GATEWAY environment variable
+
+When there is no command line argument, the `IPFS_GATEWAY` environment is next. If it contains a value, it will be used as gateway.
+
+### Gateway file
+
+**This feature is only for IPFS implementers, not for IPFS application integrations.**
+
+When the implementation provides a gateway (and it's not disabled through other means) then it should make that known to other local applications. The gateway file serves this purpose. It's a file with only 1 single line containing the full http URL to your gateway. For example, it could contain the line: "http://localhost:8080".
+
+The file conditions:
+    1. is named "gateway"
+    2. **only** exists when your implementation actually starts a gateway
+    3. is removed when your implementation shuts down
+
+For historical and compatibility readons, this file shall be placed in:
+$HOME/.ipfs/ thus the resulting end path is going to be $HOME/.ipfs/gateway
+
+Future implementers must follow the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) where the gateway file will placed in:
+**$XDG_CONFIG_HOME/ipfs/gateway**
+
+The conditions for this file are the same as those in $HOME/.ipfs/gateway.
+
+In the, admitedly rare, event of running multiple IPFS implementations each hosting their own gateway. First-come-first-serve applies here. The application that created the gateway file owns it and takes care of removing it. Subsequent instances or different application should not touch the file if it's already there.
+
+### Example implementations
+
+#### ffmpeg
+
+As of ffmpeg 5.1, it implements this logic. The source for it's implementation can be found [here](https://github.com/FFmpeg/FFmpeg/blob/master/libavformat/ipfsgateway.c).
+
+#### curl
+
+The implementation for curl is currently a work in progress but it too follows the steps as outlined in the decision tree.
+
+#### libipfsclient
+
+A reference implementation of this spec along with more functionality to retrieve data from IPFS. This is intended to be used by applications wanting to implement IPFS support.
+
+## Test fixtures
+
+N/A
+
+## Design rationale
+
+To simplify IPFS integrations in third party applications it's important to know if said application can use a gateway. This spec defines the rules to find such a gateway and/or how to influence it.
+
+### User benefit
+
+Users as in applications using IPFS. They get a defined way to find out if their host pc runs an IPFS gateway. Without this spec there is no defined way.
+
+### Compatibility
+
+Kubo currently makes `$HOME/.ipfs/gateway`, this should stay for backwards compatibility.
+A new version implementing this spec should create `$XDG_CONFIG_HOME/ipfs/gateway` in exactly the same fashion.
```

Comment:
If we want to truly enable multiple IPFS implementations, Kubo-specific config needs to go.
```suggestion
Kubo currently makes `$HOME/.ipfs/gateway`, this should stay for backwards compatibility.
A new version implementing this spec should create `$XDG_CONFIG_HOME/ipfs/gateway` in exactly the same fashion.
Future version of Kubo will switch to `$XDG_CONFIG_HOME/ipfs/gateway` and create a symlink at `$HOME/.ipfs/gateway` for backward-compatibility.
```